### ISSUE TITLE
 How to make the Image label rotate separately in svg

### ISSUE THREAD

**Issue by gh357550318** [react-native-community/react-native-svg#940]
```jsx
import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Animated, Easing
} from 'react-native';

import Svg,{
  Text,
  Image,
} from 'react-native-svg';

const AnimatedSvg = Animated.createAnimatedComponent(Svg);

export default class App extends Component {
  constructor() {
    super();
    this.RotateValueHolder = new Animated.Value(0);
  }

  componentDidMount() {
    this.StartImageRotateFunction();
  }


  StartImageRotateFunction() {
    this.RotateValueHolder.setValue(0);
    Animated.timing(this.RotateValueHolder, {
      toValue: 1,
      duration: 10000,
      easing: Easing.linear,
      useNativeDriver:true
    }).start(()=>this.StartImageRotateFunction());
  }

  render() {
    return (
      <View style={styles.container}>
        <AnimatedSvg
        width="300"
        height="300"
        style={{
          transform: [
            {
              rotate: this.RotateValueHolder.interpolate({
                inputRange: [0, 1],
                outputRange: ['0deg', '360deg'],
              }),
            },
          ],
        }}
      >
      <Image
        width="100%"
        height="100%"
        href={require('./src/assets/bg.png')}
      />
      <Text
        x="50"
        y="50"
        textAnchor="middle"
        fontWeight="bold"
        fontSize="16"
        fill="blue"
    >HOGWARTS</Text>
      </AnimatedSvg>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },

});

```

How to make the iamge tag rotate alone    not complete <Svg>？
I have tried 
const AnimatedImage = Animated.createAnimatedComponent(Image);
and put <svg> style move this const    it’s not working

**Comment by gh357550318** [comment]
@msand    please help me ,thank you

**Comment by msand** [comment]
How about something like this: https://snack.expo.io/@msand/funny-cheese

```jsx
import React, { Component } from 'react';
import { StyleSheet, View, Animated, Easing, Platform } from 'react-native';

import { Svg } from 'expo';
const { G, Text, Image, Rect } = Svg;

const AnimatedSvg = Animated.createAnimatedComponent(Svg);

const AnimatedG = Animated.createAnimatedComponent(G);

export default class App extends Component {
  constructor() {
    super();
    this.RotateValueHolder = new Animated.Value(0);
  }

  componentDidMount() {
    this.StartImageRotateFunction();
  }

  StartImageRotateFunction() {
    this.RotateValueHolder.setValue(0);
    Animated.timing(this.RotateValueHolder, {
      toValue: 1,
      duration: 10000,
      easing: Easing.linear,
      useNativeDriver: true,
    }).start(() => this.StartImageRotateFunction());
  }

  render() {
    const width = 300;
    const offsetAndroid = Platform.OS === 'android' ? width / 2 : 0;
    const [pivotX, pivotY] = [50, 50];
    return (
      <View style={styles.container}>
        <AnimatedSvg
          width={width}
          height={width}
          viewBox={`-${pivotX} -${pivotY} 100 100`}>
          <G transform={`translate(-${pivotX} -${pivotY})`}>
            <Rect width="100" height="100" fill="green" />
            <Image
              width="100%"
              height="100%"
              href={require('./assets/snack-icon.png')}
            />
          </G>
          <AnimatedG
            style={{
              transform: [
                { translateX: -offsetAndroid },
                {
                  rotate: this.RotateValueHolder.interpolate({
                    inputRange: [0, 1],
                    outputRange: ['0deg', '360deg'],
                  }),
                },
                { translateX: offsetAndroid },
              ],
            }}>
            <Rect width="20" height="20" fill="blue" />
            <Text
              textAnchor="middle"
              fontWeight="bold"
              fontSize="16"
              fill="blue">
              HOGWARTS
            </Text>
          </AnimatedG>
        </AnimatedSvg>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});
```

Based on: https://github.com/react-native-community/react-native-svg/issues/857

**Comment by gh357550318** [comment]
how is 'expo'?  
Why not is  react-native-svg？
‘expo ’  Where to get it from

**Comment by gh357550318** [comment]
wow  i know  
your react-native-svg version is？

**Comment by gh357550318** [comment]
![bg](https://user-images.githubusercontent.com/44966458/52942716-76122880-33a6-11e9-8007-d76014058ba1.png)

For example, if the picture is round, let the picture rotate at this center of mind .

**Comment by gh357550318** [comment]
@msand
 style={{
              transform: [
                { translateX: 150 },
                {
                  rotate: this.RotateValueHolder.interpolate({
                    inputRange: [0, 1],
                    outputRange: ['0deg', '360deg'],
                  }),
                },
                { translateX: 150 },
              ],
            }}


What is controlling the center of rotation? 
translateX ？

**Comment by msand** [comment]
In that case it is in the origin

**Comment by msand** [comment]
@gh357550318 Perhaps this can help you:

```jsx
import React, { Component } from 'react';
import { StyleSheet, View, Animated, Easing, Platform } from 'react-native';

import { Svg, G, Text, Rect } from 'react-native-svg';

const AnimatedG = Animated.createAnimatedComponent(G);

function StaticExample() {
  const angle = 45;
  const x = 50;
  const y = 50;
  return (
    <Svg width="30%" height="30%" viewBox="0 0 100 100">
      <Rect width="100" height="100" fill="green" />
      <G transform={`rotate(${angle}, ${x}, ${y})`}>
        <Rect x={50} y={50} width="20" height="20" fill="blue" />
        <Text
          x={x}
          y={y}
          textAnchor="middle"
          fontWeight="bold"
          fontSize="16"
          fill="blue"
        >
          No animation
        </Text>
      </G>
    </Svg>
  );
}

class JSAnimation extends Component {
  RotateValueHolder = new Animated.Value(0);

  componentDidMount() {
    Animated.loop(
      Animated.timing(this.RotateValueHolder, {
        toValue: 1,
        duration: 10000,
        easing: Easing.linear,
        useNativeDriver: false,
        repeat: true,
      }),
    ).start();
  }

  render() {
    return (
      <Svg width="30%" height="30%" viewBox="0 0 100 100">
        <Rect width="100" height="100" fill="green" />
        <G transform="translate(50, 50)">
          <AnimatedG
            transform={this.RotateValueHolder.interpolate({
              inputRange: [0, 1],
              outputRange: ['rotate(0)', 'rotate(360)'],
            })}
          >
            <G transform="translate(-50, -50)">
              <Rect x={50} y={50} width="20" height="20" fill="blue" />
              <Text
                x={50}
                y={50}
                textAnchor="middle"
                fontWeight="bold"
                fontSize="14"
                fill="blue"
              >
                JS Animation
              </Text>
            </G>
          </AnimatedG>
        </G>
      </Svg>
    );
  }
}

class NativeAnimation extends Component {
  RotateValueHolder = new Animated.Value(0);

  componentDidMount() {
    Animated.loop(
      Animated.timing(this.RotateValueHolder, {
        toValue: 1,
        duration: 10000,
        easing: Easing.linear,
        useNativeDriver: true,
        repeat: true,
      }),
    ).start();
  }

  render() {
    const width = 300;
    const offsetAndroid = Platform.OS === 'android' ? width / 2 : 0;
    return (
      <View style={styles.container}>
        <StaticExample />
        <Svg width={width} height={width} viewBox={`-50 -50 100 100`}>
          <Rect x="-50" y="-50" width="100" height="100" fill="green" />
          <AnimatedG
            style={{
              transform: [
                { translateX: -offsetAndroid },
                {
                  rotate: this.RotateValueHolder.interpolate({
                    inputRange: [0, 1],
                    outputRange: ['0deg', '360deg'],
                  }),
                },
                { translateX: offsetAndroid },
              ],
            }}
          >
            <Rect width="20" height="20" fill="blue" />
            <Text
              textAnchor="middle"
              fontWeight="bold"
              fontSize="14"
              fill="blue"
            >
              Native animation
            </Text>
          </AnimatedG>
        </Svg>
      </View>
    );
  }
}

export default class App extends Component {
  render() {
    return (
      <View style={styles.container}>
        <StaticExample />
        <JSAnimation />
        <NativeAnimation />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});

```

**Comment by msand** [comment]
The rotations happen about the origin, so if you want to rotate in some other coordinate system, you first need to transform that one to the origin, apply the rotation, and transform it back. That is what the spec says that the optional arguments to the rotate transform definition must do:
https://www.w3.org/TR/SVG11/coords.html#TransformAttribute
> `rotate(<rotate-angle> [<cx> <cy>])`, which specifies a rotation by `<rotate-angle>` degrees about a given point.
> 
> If optional parameters `<cx>` and `<cy>` are not supplied, the rotate is about the origin of the current user coordinate system. The operation corresponds to the `matrix [cos(a) sin(a) -sin(a) cos(a) 0 0]`.
> 
> If optional parameters `<cx>` and `<cy>` are supplied, the rotate is about the point `(cx, cy)`. The operation represents the equivalent of the following specification: `translate(<cx>, <cy>) rotate(<rotate-angle>) translate(-<cx>, -<cy>)`.

More info at:
https://www.w3.org/TR/css-transforms-1/#svg-transform-functions

If you have many of these, I would recommend moving the origin to the center, by using 
`viewBox="-originX -originY width height"` as I've done in the native animation example (with a width and height of 100 units to make them similar to percentages, shifted half of that i.e., 50 units in both directions). And, rendering the content such that the rotation doesn't need any additional transforms, but can be animated in-place instead.

**Comment by msand** [comment]
@gh357550318 updated the example a bit

**Comment by gh357550318** [comment]
I am going to try，

I am very honored to meet you.
@msand

**Comment by gh357550318** [comment]
error while updating property 'transform' of a view managed by : RNSVGGroup

Value for transform cannot be cast from String to ReadableArray ?? 

what happend??

dear

**Comment by gh357550318** [comment]
@msand 
i'm sorry ！
I still didn't succeed.

`import React, { Component } from 'react';
import { Animated, Easing } from 'react-native';
import Svg, { G, Text, Rect,Image } from 'react-native-svg';

const AnimatedSvg = Animated.createAnimatedComponent(Svg);
export default class CutterSvg extends Component {
  constructor() {
    super();
    this.RotateValueHolder = new Animated.Value(0);
 componentDidMount() {
    this.StartImageRotateFunction();
  }

  StartImageRotateFunction() {
    this.RotateValueHolder.setValue(0);
    Animated.timing(this.RotateValueHolder, {
      toValue: 1,
      duration: 10000,
      easing: Easing.linear,
      useNativeDriver: true,
    }).start(() => this.StartImageRotateFunction());
  }
render() {
    const { tbmData }= this.props
    const spin = this.RotateValueHolder.interpolate({
      inputRange:[0,1],
      outputRange:['0deg','360deg']
    })
      return (
        <AnimatedSvg
          height='100%'
          width='100%'
          viewBox={`0 0 500 500`}
          >
          <Image
              x='-50'
              y='20'
              width="85%"
              height="85%"
              // style={{transform:[{rotate:spin}]}}
              href={require('../../assets/images/bg.png')}
            />
          {/* four text */}
          {this.textData(tbmData)}
        </AnimatedSvg>
 
    );
  }

`

`react-native 0.58.4`
`react-native-svg 9.2.4`
![_20190228193435](https://user-images.githubusercontent.com/44966458/53563900-6954b800-3b90-11e9-81da-0ae0ffd7430e.png)

i want the images rotate  on origin  360°

**Comment by msand** [comment]
Can you try with e.g. the latest commit from the [vector-effect branch](https://github.com/react-native-community/react-native-svg/tree/vector-effect) e.g.:
`yarn add react-native-svg@react-native-community/react-native-svg#dc94707`

**Comment by gh357550318** [comment]
This is not the answer I want.  oh

**Comment by msand** [comment]
Btw, if you're only rendering png images and text, why not use the react-native Image and Text components? And react-native style transforms on that? It's likely to be more efficient anyway.

**Comment by gh357550318** [comment]
i had used react-native Image and Animation,
but this Images cover my svgrect and svgtext.
maybe i need zIndex, but use zIndex not succeed,
@msand

**Comment by msand** [comment]
Should be enough to have the image render before the text in document order. I'll take a look at your example when I have a bit less urgent schedule.

**Comment by gh357550318** [comment]
ok 
Express thankfulness

**Comment by juandiegombr** [comment]
Hello.

I arrived to this question after trying to animate the rotation of a Path in a Svg. I have tried the examples above in the expo playground.

I have noticed a difference between using Svg from expo and from 'react-native-svg'. In my plain react-native project I got the rotate animation in a Svg component but not in any of the children. Without expo, AnimatedG or any other Animated component besides AnimatedSvg is not working to me.
